src/main/java/frc/robot/subsystems/drivetrain/SwerveDrive.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems.drivetrain;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.wpilibj.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.Robot;
+
+import static frc.robot.Constants.SwerveDrive.*;
+
+public class SwerveDrive extends SubsystemBase {
+    private Timer timer = new Timer();
+    private double currentTime;
+    private double lastTime = 0;
+    private final SwerveModule[] modules = new SwerveModule[4];
+    private final SwerveDriveKinematics kinematics = new SwerveDriveKinematics(
+            new Translation2d(Ry, Rx),
+            new Translation2d(Ry, -Rx),
+            new Translation2d(-Ry, Rx),
+            new Translation2d(-Ry, -Rx)
```

Comment:
Wrong axis (first of all, you need to pass (x, y) not (y, x) and check your +- again)